### PR TITLE
Added support for getting list of workspaces

### PR DESCRIPTION
**Pull request by fkrafi**

### CODE REVIEW

**Comment by fkrafi** [comment]
@aubustou @beelit94 , can you please review this PR

**Comment by fkrafi** [comment]
@Spikeophant @beelit94 and @@aubustou , can you please review this PR

File: test/test_terraform.py [hunk 1]
```diff
@@ -447,3 +447,9 @@ def test_delete_workspace_with_args(self, workspace_setup_teardown, caplog):
             f"Command: terraform workspace delete -force test {current_path}"
             in caplog.messages
         )
+
+    def test_list_workspace(self):
+        tf = Terraform(working_dir=current_path)
+        workspaces = tf.list_workspace()
+        assert len(workspaces) > 0
+        assert 'default' in workspaces
```
Comment (beelit94) [inline test/test_terraform.py]: same here, pylint compliance

File: python_terraform/terraform.py [hunk 2]
```diff
@@ -456,6 +456,13 @@ def show_workspace(self, **kwargs) -> CommandOutput:
         """
         return self.cmd("workspace", "show", **kwargs)
 
+    def list_workspace(self) -> List[str]:
+        """List of workspaces
+        
+        :return: workspaces
+        """
+        return list(filter(lambda x: len(x) > 0, map(lambda x: x.strip('*').strip(), self.cmd("workspace", "list")[1].split())))
```
Comment (beelit94) [inline python_terraform/terraform.py]: please change format to pylint compliant.
Could you also give an example what would output of workspace looks like?

I'm worried about potential "KeyError" or "indexError" from your hard coded `1` here

File: test/test_terraform.py (hunk 1 above)
Comment (fkrafi) [reply test/test_terraform.py]: used pycodestyle as pylint

File: python_terraform/terraform.py (hunk 2 above)
Comment (fkrafi) [reply python_terraform/terraform.py]: - used pycodestyle as pylint
- self.cmd returns will always be size 3 but 2nd one can be None. fixed that with checking, so no more KeyError or any other error.

File: python_terraform/terraform.py [hunk 3]
```diff
@@ -456,6 +456,25 @@ def show_workspace(self, **kwargs) -> CommandOutput:
...
+        :example:
+            >>> tf = Terraform()
+            >>> tf.list_workspace()
+            ['default', 'test']
+        """
+        return list(
+            filter(
+                lambda x: len(x) > 0,
+                map(
+                    lambda x: x.strip('*').strip(),
```
Comment (beelit94) [inline python_terraform/terraform.py]: sorry, one more NIT item, could you rename the `x` , and give it a meaningful name?

**Comment by fkrafi** [comment]
@beelit94 @aubustou @BNMetrics @raquel-ucl @0x29a @xishengcai 
Can you please verify this PR,

**Comment by fkrafi** [comment]
@kirankotari , can you please look into this PR.

**Comment by kirankotari** [comment]
> @kirankotari , can you please look into this PR.

Give me a day or two to review.

**Review by kirankotari — APPROVED** [review]

**Pull request merged**